.github/workflows/ci.yml

```diff
@@ -0,0 +1,443 @@
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: 3.3
+          bundler-cache: true
+
+      - name: Write used versions into file
+        run: bundle exec rake ports:version_file
+
+      - name: Cache ports
+        uses: actions/cache@v4
+        with:
+          path: ports
+          key: native-v2-${{ hashFiles('**/.ports_versions') }}
+          restore-keys: |
+            native-${{ hashFiles('* */.ports_versions') }}
+            native-v2-
+
+      - name: Build required libraries
+        run: |
+          bundle exec rake ports
+
+  test-linux:
+    needs:
+      - compile-native-ports
+    strategy:
+      matrix:
+        mssql-version:
+          - 2017
+          - 2019
+          - 2022
+        ruby-version:
+          - 2.7
+          - '3.0'
+          - 3.1
+          - 3.2
+          - 3.3
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby-version }}
+          bundler-cache: true
+
+      - name: Write used versions into file
+        run: |
+          bundle exec rake ports:version_file
+
+      - name: Cache ports
+        uses: actions/cache@v4
+        with:
+          path: ports
+          key: native-v2-${{ hashFiles('**/.ports_versions') }}
+          fail-on-cache-miss: true
+
+      - name: Build gem
+        run: |
+          bundle exec rake build
+
+      - name: Setup MSSQL
+        uses: potatoqualitee/mssqlsuite@v1.7
+        with:
+          install: sqlengine, sqlclient
+          version: ${{ matrix.mssql-version }}
+          sa-password: "c0MplicatedP@ssword"
+          show-log: true
+
+      - name: Setup MSSQL database
+        run: |
+          sqlcmd -S localhost -U sa -P "c0MplicatedP@ssword" -i ./test/sql/db-create.sql
+          sqlcmd -S localhost -U sa -P "c0MplicatedP@ssword" -i ./test/sql/db-login.sql
+
+      - name: Install toxiproxy-server
+        run: |
+          wget -O toxiproxy-2.5.0.deb https://github.com/Shopify/toxiproxy/releases/download/v2.5.0/toxiproxy_2.5.0_linux_amd64.deb
+          sudo dpkg -i toxiproxy-2.5.0.deb
+          sudo toxiproxy-server &
+
+      - name: Run tests
+        run: bundle exec rake test
+        env:
+          TOXIPROXY_HOST: "localhost"
+
+      - name: Test Summary
+        uses: test-summary/action@v2
+        with:
+          paths: "test/reports/TEST-*.xml"
+        if: always()
+
+  install_macos:
+    strategy:
+      fail-fast: false
+      matrix:
+        ruby-version:
+          - 2.7
+          - "3.0"
+          - 3.1
+          - 3.2
+          - 3.3
+
+    name: Install on Mac OS (M1)
+    runs-on: macos-14
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install FreeTDS
+        run: brew install freetds
+        shell: bash
+
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby-version }}
```

Comment:
Do we need bundler-cache: true on all these? Why the difference?

Reply:
I think was just an oversight. I added it now to all jobs: 49c557d0.